tst/unit/test_index_split.cpp

```diff
@@ -0,0 +1,157 @@
+
+    Metadata m({Metadata::Independent, Metadata::WithFluxes}, scalar_shape);
+    Metadata m_vector({Metadata::Independent, Metadata::WithFluxes, Metadata::Vector},
+                      vector_shape);
+    auto pkg = std::make_shared<StateDescriptor>("Test package");
+    pkg->AddField(v1::name(), m);
+    pkg->AddField(v3::name(), m_vector);
+    pkg->AddField(v5::name(), m);
+    BlockList_t block_list = MakeBlockList(pkg, NBLOCKS, N, NDIM);
+
+    MeshData<Real> mesh_data("base");
+    mesh_data.Set(block_list, nullptr, NDIM);
+
+    WHEN("We initialize an IndexSplit with all outer k and no outer j") {
+      IndexSplit sp(&mesh_data, IndexDomain::interior, IndexSplit::all_outer,
+                    IndexSplit::no_outer);
+      THEN("The outer range should be appropriate") { REQUIRE(sp.outer_size() == N); }
+      THEN("The inner ranges should be appropriate") {
+        using atomic_view = Kokkos::MemoryTraits<Kokkos::Atomic>;
+        Kokkos::View<int *, atomic_view> nwrong("nwrong", 1);
+        parthenon::par_for_outer(
+            DEFAULT_OUTER_LOOP_PATTERN, "Test IndexSplit", DevExecSpace(), 0, 0, 0,
+            sp.outer_size() - 1,
+            KOKKOS_LAMBDA(parthenon::team_mbr_t member, const int outer_idx) {
+              const auto krange = sp.GetBoundsK(outer_idx);
+              const auto jrange = sp.GetBoundsJ(outer_idx);
+              const auto irange = sp.GetInnerBounds(jrange);
+              // JMM: Note that these are little cleaner without ghosts
+              if (!(krange.s == outer_idx)) nwrong(0) += 1;
+              if (!(krange.e == outer_idx)) nwrong(0) += 1;
+              if (!(jrange.s == 0)) nwrong(0) += 1;
+              if (!(jrange.e == N - 1)) nwrong(0) += 1;
+              if (!(irange.s == 0)) nwrong(0) += 1;
+              if (!(irange.e == (N * N - 1))) nwrong(0) += 1;
+            });
+        auto nwrong_h = Kokkos::create_mirror_view(nwrong);
+        Kokkos::deep_copy(nwrong_h, nwrong);
+        REQUIRE(nwrong_h(0) == 0);
+      }
+    }
+    WHEN("We initialize an IndexSplit with outer k and outer j") {
+      IndexSplit sp(&mesh_data, IndexDomain::interior, IndexSplit::all_outer,
+                    IndexSplit::all_outer);
+      THEN("the outer index range should be appropriate") {
+        REQUIRE(sp.outer_size() == (N * N));
+      }
+      THEN("The inner index ranges should be appropriate") {
+        using atomic_view = Kokkos::MemoryTraits<Kokkos::Atomic>;
+        Kokkos::View<int *, atomic_view> nwrong("nwrong", 1);
+        parthenon::par_for_outer(
+            DEFAULT_OUTER_LOOP_PATTERN, "Test IndexSplit", DevExecSpace(), 0, 0, 0,
+            sp.outer_size() - 1,
+            KOKKOS_LAMBDA(parthenon::team_mbr_t member, const int outer_idx) {
+              const auto krange = sp.GetBoundsK(outer_idx);
+              const auto jrange = sp.GetBoundsJ(outer_idx);
+              const auto irange = sp.GetInnerBounds(jrange);
```

Comment:
`krange` and `jrange` could be computed here to check, right?

Reply:
Done.